homeassistant/components/mqtt/mixins.py

```diff
@@ -674,6 +690,14 @@ def __init__(
             discovery_hash,
         )
 
+    @callback
```

Comment:
Since these are not entities, the platforms are not loaded, so the cleanup takes place here